### PR TITLE
Add in Javadocs and JUnit testing

### PR DESCRIPTION
**Pull request by kyhjonathan**
Yet to fix issue #131.
Will be fixed asap, along with further documentation and testing

### CODE REVIEW

**Review by clarencepohh — COMMENTED** [review]
Tests look good, as do the checkstyle changes and comments. 
Likely have to fix some merge issues

**Comment by kyhjonathan** [comment]
Fixed issue #131

File: src/main/java/data/TaskManager.java [hunk 1]
```diff
@@ -185,6 +199,16 @@ private static Task updateEventTask(Scanner scanner, List<Task> dayTasks,
         }
     }
 
+    /**
```
Comment (Mahesh1772) [inline src/main/java/data/TaskManager.java:202]: Good job adding JavaDocs for methods which was missed previously.

File: src/main/java/data/Event.java [hunk 2]
```diff
@@ -100,6 +100,7 @@ public String getSaveFormat () {
                 + getEndTime();
     }
 
+    //@@author kyhjonathan-unused
```
Comment (Aak242) [inline src/main/java/data/Event.java:103]: This can be moved to the unused folder as mentioned by Clarence. In addition, we can follow the same folder naming standard as our actual code. Nicholas has placed his unused classes in a ui folder within unused. We should similarly add our unused code belonging to a certain package into a folder with the same name within the unused folder.

File: src/main/java/data/TaskManager.java [hunk 3]
```diff
@@ -159,6 +164,16 @@ public static void updateTask(LocalDate date, int taskIndex, String newTaskDescr
...
+     * Updates the details of an Event task.
+     *
+     * @param scanner Scanner object to read user input for updating task details.
+     * @param dayTasks List of tasks for the day.
+     * @param taskIndex Index of the task to be updated.
+     * @param newTaskDescription New description for the task.
+     * @param oldDescription Old description of the task.
+     * @return Updated Task object.
+     */
```
Comment (Aak242) [inline src/main/java/data/TaskManager.java:176]: Concise but clear documentation!

File: src/main/java/data/TaskManager.java [hunk 4]
```diff
@@ -751,6 +828,7 @@ public List<String> getFreeTimeSlots(List<Task> events, LocalDate currentDate) {
             }
         });
 
+        // Initialize the last end time to the start of the day for the first day
         // Initialize the last end time to the start of the day for the first day
```
Comment (Mahesh1772) [inline src/main/java/data/TaskManager.java:832]: Looks to be the same comment twice!

File: src/main/java/data/TaskManager.java [hunk 5]
```diff
@@ -568,20 +628,19 @@ public static TaskType parseTaskType(String userInput) {
...
-     * @param scanner User input.
-     * @param weekView Current week being viewed.
-     * @param monthView Current month being viewed.
-     * @param inMonthView Whether month is being viewed.
-     * @param taskManager The taskManager class being used.
+     * @param scanner Scanner object to read user input for task details.
+     * @param weekView WeekView object for finding the date.
+     * @param monthView MonthView object for finding the date.
+     * @param inMonthView A boolean indicating whether the view is in month view or not.
+     * @param taskManager The TaskManager instance being used.
```
Comment (Aak242) [inline src/main/java/data/TaskManager.java:637]: Thanks for improving the clarity of the JavaDocs!

File: src/test/java/data/TaskManagerExceptionTest.java [hunk 6]
```diff
@@ -31,6 +34,89 @@ public void checkIfDateInCurrentMonth_pastDateGiven_exceptionThrown () {
...
+    public void checkIfTimeInFormat_invalidTimeGiven_exceptionThrown () {
+        TaskManagerException thrown = Assertions.assertThrows(TaskManagerException.class, () ->
+                checkIfTimeInFormat("abcde"),
+                "TaskManagerException was expected");
+
+        Assertions.assertEquals("Invalid time format. Please use the format HHmm", thrown.getMessage());
+    }
+
+    @Test
+    void checkIfTimeInFormat_invalidFormat_throwsException() {
```
Comment (Mahesh1772) [inline src/test/java/data/TaskManagerExceptionTest.java:47]: Can we use "_" while names methods?

File: src/main/java/storage/Storage.java [hunk 7]
```diff
@@ -120,6 +128,7 @@ public static Map<LocalDate, List<Task>> loadTasksFromFile(Path path) {
         logger.log(Level.INFO, "tasks returned");
         return tasks;
     }
+    //@@author
```
Comment (Aak242) [inline src/main/java/storage/Storage.java:131]: Is this meant to be attributed to someone?

**Review by Mahesh1772 — COMMENTED** [review]
LGTM! Thanks for adding extensive testing and Javadocs

File: src/test/java/data/TaskManagerExceptionTest.java [hunk 8]
```diff
@@ -31,6 +34,89 @@ public void checkIfDateInCurrentMonth_pastDateGiven_exceptionThrown () {
...
+        } catch (TaskManagerException e) {
+            assert false;
+        }
+    }
+
+    @Test
+    void checkIfTimeInFormat_invalidMinute_throwsException() {
+        String invalidTime = "23:60";
+        assertThrows(TaskManagerException.class, () -> TaskManagerException.checkIfValidTime(invalidTime));
+    }
```
Comment (Aak242) [inline src/test/java/data/TaskManagerExceptionTest.java:66]: Good job on adding extensive tests and also on testing both positive and negative test cases!

**Review by Aak242 — COMMENTED** [review]
Apart from moving the unused code into the folder Nicholas created, everything else LGTM!

File: docs/UserGuide.md [hunk 9]
```diff
@@ -361,7 +361,7 @@ Exiting Calendar...
 ```
 The application will exit after displaying a goodbye message.
 
-### ICS Exporting and Importing: `ics` `[coming in v2.1]`
+### ICS Exporting and Importing: `ics` `[coming in future versions]`
```
Comment (NicholasTanYY) [inline docs/UserGuide.md:364]: Good spot on this! We should definitely leave the ICS handling to future versions :)

File: src/main/java/data/exceptions/TaskManagerException.java [hunk 10]
```diff
@@ -52,6 +52,17 @@ public static void checkIfDateInFormat(String date) throws TaskManagerException
...
+     * need to debug this method
+     */
+    public static void checkIfTaskExistsInCurrentDate(List<Task> dayTasks, int index) throws TaskManagerException {
+        if (index <= 0) {
+            throw new TaskManagerException("Invalid task number. Please try again.");
+        } else if (index > dayTasks.size()) {
+            throw new TaskManagerException("Task number does not exist. Please try again.");
+        }
+    }
+
```
Comment (NicholasTanYY) [inline src/main/java/data/exceptions/TaskManagerException.java:65]: Great handling of the case where the user might accidentally enter a wrong index!

File: src/test/java/data/TaskManagerExceptionTest.java [hunk 11]
```diff
@@ -31,6 +34,89 @@ public void checkIfDateInCurrentMonth_pastDateGiven_exceptionThrown () {
...
+
+    @Test
+    void checkIfTimeInFormat_validFormat_noExceptionThrown() {
+        String validTime = "23:59";
+        try {
+            TaskManagerException.checkIfValidTime(validTime);
+        } catch (TaskManagerException e) {
+            assert false;
+        }
+    }
```
Comment (NicholasTanYY) [inline src/test/java/data/TaskManagerExceptionTest.java:60]: Great check to see if the date is in the correct format!

File: unused/icshandler/IcsHandler.java [hunk 12]
```diff
@@ -16,6 +16,10 @@
 import java.io.IOException;
 import java.util.GregorianCalendar;
 
+//@@author kyhjonathan-unused
+//Unused as there were too many issues with the dependacies of the library
+//and the library itself was not very well documented
+//Decided to half development of ics and focuses on testing and documentation of other features
 public class IcsHandler {
```
Comment (NicholasTanYY) [inline unused/icshandler/IcsHandler.java:23]: yes, you should definitely be credited for attempting to incorporate the ICS handler!

**Review by NicholasTanYY — APPROVED** [review]
Overall , great job on writing many test cases and placing the ICS handler into the unused documents! LGTM

File: src/main/java/data/TaskManager.java [hunk 13]
```diff
@@ -138,19 +137,19 @@ public static void updateTask(LocalDate date, int taskIndex, String newTaskDescr
...
+                            oldDescription + " to: " + newTaskDescription);
+                    break;
+                case "E":
+                    task = updateEventTask(scanner, dayTasks, taskIndex, newTaskDescription, oldDescription);
+                    break;
+                case "D":
+                    task = updateDeadlineTask(scanner, dayTasks, taskIndex, newTaskDescription, oldDescription);
+                    break;
+                default:
+                    throw new IllegalArgumentException("Invalid task type");
```
[the remark this answers is not in the document]
Comment (kyhjonathan) [reply src/main/java/data/TaskManager.java]: I've amended the indentation. Thanks for spotting the mistake!
